src/main/java/glowredman/amazingtrophies/model/complex/BaseModelStructure.java

```diff
@@ -176,18 +178,18 @@ private void removeTransparentBlocks(String[][] structure, Set<Character> transp
         }
     }
 
-    private Set<Character> getTransparentBlocks() {
-        Set<Character> transparentBlocks = new HashSet<>();
+    private CharSet getTransparentBlocks() {
+        CharSet transparentBlocks = new CharOpenHashSet();
 
         // Iterate over all blocks to find transparent ones.
-        for (Map.Entry<Character, Pair<Block, Integer>> entry : charToBlock.entrySet()) {
+        for (Char2ObjectMap.Entry<BlockMeta> entry : charToBlock.char2ObjectEntrySet()) {
```

Comment:
This allocates a new Char2ObjectMap.Entry object for each iteration of the loop, you can use the .fastIterator() method on the fast entry set with a manual for loop or .fastForEach() with a lambda, both will end up with just 1-2 allocations per loop

Reply:
Done, thanks for the advice!